filament/backend/src/HandleAllocator.cpp

```diff
@@ -24,26 +24,29 @@ namespace filament::backend {
 
 using namespace utils;
 
+template <int P0, int P1, int P2>
 UTILS_NOINLINE
-HandleAllocator::Allocator::Allocator(AreaPolicy::HeapArea const& area)
+HandleAllocator<P0, P1, P2>::Allocator::Allocator(AreaPolicy::HeapArea const& area)
         : mArea(area) {
     // TODO: we probably need a better way to set the size of these pools
     const size_t unit = area.size() / 32;
     const size_t offsetPool1 =      unit;
     const size_t offsetPool2 = 16 * unit;
     char* const p = (char*)area.begin();
```

Comment:
We probably also need some parameters to define the relative size of each pool arena, can be in another PR.